boto3/session.py

```diff
@@ -78,7 +81,10 @@ def __init__(
 
         if aws_access_key_id or aws_secret_access_key or aws_session_token:
```

Comment:
Since we do it with the other credential params, should we set credentials if account ID is set but none of the others are?